### ISSUE TITLE
node-red-node-pi-gpiod | PWM Select frequency and precision 

### ISSUE THREAD

**Issue by navivfr** [node-red/node-red-nodes#746]
2 points :

**PWM @100 is not 100% DUTY**
if I set the msg.payload to 100. In the oscilloscope I dont get a 'flat' output. More like a 99% duty.

**FREQUENCY SELECTION : How to change it?**
Frequency seem to be set a 800Hz. There is a way to change it (ie 2Khz)
REM : on python I can set frequency (I tested 42Khz without any issue), and the Duty precision is crazy (Nice Dirac peak)
From 0 to 1.000.000 (100% duty)

  Node-RED version: 1.2.7
  node.js version: 12.18.2
  npm version: 6.14.18
  Platform/OS: PI 4 4gb/ Ubuntu 20.10
  Browser: Firefox

**Comment by dceejay** [comment]
ok

so we set the frequency to `PiGPIO.set_PWM_dutycycle(node.pin, parseInt(out * 2.55));`
and docs says -

    int gpioPWM(unsigned user_gpio, unsigned dutycycle)
    Starts PWM on the GPIO, dutycycle between 0 (off) and range (fully on). Range defaults to 255.

so 100 * 2.55 should = 255.

Indeed no-one has implemented the frequency setting .
From the docs - http://abyz.me.uk/rpi/pigpio/cif.html -  it looks like there are only certain suitable values
(default sample rate is 5uS)

```
The selectable frequencies depend upon the sample rate which may be 1, 2, 4, 5, 8, or 10 microseconds (default 5).

The frequencies for each sample rate are:

                       Hertz

       1: 40000 20000 10000 8000 5000 4000 2500 2000 1600
           1250  1000   800  500  400  250  200  100   50

       2: 20000 10000  5000 4000 2500 2000 1250 1000  800
            625   500   400  250  200  125  100   50   25

       4: 10000  5000  2500 2000 1250 1000  625  500  400
            313   250   200  125  100   63   50   25   13
sample
 rate
 (us)  5:  8000  4000  2000 1600 1000  800  500  400  320
            250   200   160  100   80   50   40   20   10

       8:  5000  2500  1250 1000  625  500  313  250  200
            156   125   100   63   50   31   25   13    6

      10:  4000  2000  1000  800  500  400  250  200  160
            125   100    80   50   40   25   20   10    5
```

**Comment by navivfr** [comment]
if can help, from this thread look at the reply :
https://raspberrypi.stackexchange.com/questions/107155/is-it-possible-to-get-12-bit4096-resolution-output-from-pi-hardware-pwm-pin

_My pigpio Python module will give a million steps for frequencies less than 250 Hz.

pigpio will give a resolution of 250 million divided by the frequency in hertz that you want to use.

See hardware_PWM_

```
#!/usr/bin/env python

import time
import pigpio

PWM=18

FREQ=5000

pi = pigpio.pi()

if not pi.connected:
   exit()

for percent in range(100): # 0-100 percent
   pi.hardware_PWM(PWM, FREQ, percent * 1000000 / 100)
   time.sleep(0.2)

pi.hardware_PWM(PWM, 0, 0) # stop PWM

pi.stop()
```

**Comment by dceejay** [comment]
we don't use that python library so not available to us.
Will see if I can add the choices from above.

**Comment by dceejay** [comment]
published as version 0.2.0

**Comment by navivfr** [comment]
Thx a lot... great job :-)
